### ISSUE TITLE
Potential false positive for `undocumented-param (D417)`

### ISSUE THREAD

**Issue by harupy** [astral-sh/ruff#13414]
I have the following function. The docstring has a nested `Example` section in the `x` description.

```python
def f(x, y):
    """
    Args:
        x: This is x.

            Example:

            .. code-block:: python

                import mlflow

                mlflow.log_param("x", x)

        y: This is y.
    """
```

This function violates `D417`:

```
Missing argument description in the docstring for `f`: `y`
```

Reproduction:

https://play.ruff.rs/714c1337-edee-40c2-857a-435dd9a5d243
